internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -210,42 +214,44 @@ spec:
               key: url
       script: |
         #!/usr/bin/env bash
-        TASKRUN="/tmp/$$.sh"
+        # shellcheck disable=SC2317 # shellcheck calls all the commands in the function unreachable
```

Comment:
found it in my terminal buffer: https://gist.github.com/theflockers/4eb3b9c1e3866976e455cd60e3bf5f8d

Reply:
Thanks. Really weird. I am tryting to figure this out now - so removing it from the PR temporarily (Johnny is out today anyway, so hopefully he won't mind)

Reply:
OK, I think I know why now. It's because it doesn't see the function being called anywhere. And why not? Because it's called like this and it's confused by it:
```sh
        export -f watch_build_state
        timeout "$(params.buildTimeoutSeconds)" bash -c watch_build_state

```
So yeah, let's ignore it there. But maybe add a comment explaining it? Like:
```
# Shellcheck thinks all commands in the functions are unreachable. That's because
# this function is not called directly, but via `timeout`
```

Reply:
Amended the comment to mention the `timeout` thing

Reply:
Thanks. I am not really sure why, but somehow this also means that it doesn't do any other checks in the function. Otherwise it would have said that you should wrap `$(results.jsonBuildInfo.path)` in quotes or that you should use `$()` instead of backticks. Feel free to fix those. Up to you.

Reply:
updated